<?xml version="1.0"?>
<launch>
  <!-- Standalone example launch file -->

  <!-- For setting debug level to debug -->
  <env name="ROSCONSOLE_CONFIG_FILE"
       value="$(find microstrain_3dm_gx5_45)/config/custom_rosconsole.conf"/>

  <arg name="port" default="/dev/ttyUSB0"  />
  <arg name="baudrate" default="115200"  />
  <arg name="device_setup" default="true"  />
  <arg name="readback_settings" default="true"  />
  <arg name="save_settings" default="true"  />
  <arg name="auto_init" default="true"  />

  <arg name="gps_rate" default="4"  />
  <arg name="imu_rate" default="10"  />
  <arg name="nav_rate" default="10"  />

  <arg name="dynamics_mode" default="1"  />
  <arg name="declination_source" default="2"  />
  <arg name="declination" default="0.23"  />
  <arg name="gps_frame_id" default="wgs84"  />
  <arg name="imu_frame_id" default="base_link"  />
  <arg name="odom_frame_id" default="wgs84"  />
  <arg name="odom_child_frame_id" default="base_link"  />

  <arg name="publish_gps" default="true"  />
  <arg name="publish_imu" default="true"  />

  <!-- Microstain sensor node -->
  <node name="microstrain_3dm_gx5_45_node"
  pkg="microstrain_3dm_gx5_45"
  type="microstrain_3dm_gx5_45_node" output="screen">
    <!--<param name="port" value="/dev/ttyACM0" type="str" />-->
    <param name="port" value="$(arg port)" type="str" />
    <param name="baudrate" value="$(arg baudrate)" type="int" />

    <param name="device_setup" value="$(arg device_setup)" type="bool" />
    <param name="readback_settings" value="$(arg readback_settings)" type="bool" />
    <param name="save_settings" value="$(arg save_settings)" type="bool" />
    <param name="auto_init" value="$(arg auto_init)" type="bool" />
    <param name="gps_rate" value="$(arg gps_rate)" type="int" />
    <param name="imu_rate" value="$(arg imu_rate)" type="int" />
    <param name="nav_rate" value="$(arg nav_rate)" type="int" />
    <param name="dynamics_mode" value="$(arg dynamics_mode)" type="int" />
    <param name="declination_source" value="$(arg declination_source)" type="int" />
    <param name="declination" value="$(arg declination)" type="double" />

    <param name="gps_frame_id" value="$(arg gps_frame_id)" type="str" />
    <param name="imu_frame_id" value="$(arg imu_frame_id)" type="str" />
    <param name="odom_frame_id" value="$(arg odom_frame_id)" type="str" />
    <param name="odom_child_frame_id" value="$(arg odom_child_frame_id)" type="str" />

    <param name="publish_gps" value="$(arg publish_gps)" type="bool" />
    <param name="publish_imu" value="$(arg publish_imu)" type="bool" />
    <param name="publish_odom" value="$(arg publish_odom)" type="bool" />


    <remap from="nav/odom" to="nav_odom"/>
    <remap from="nav/status" to="nav_status"/>


  </node>

</launch>
